client/src/connection/itc/index.ts

```diff
@@ -0,0 +1,438 @@
+        this._workDirectory = parts[1].trim();
+        this._runResolve();
+        updateStatusBarItem(true);
+        return;
+      }
+      if (!this.processLineCodes(line)) {
+        this._html5FileName =
+          line.match(/NOTE: .+ HTML5.* Body.+: (.+)\.htm/)?.[1] ??
+          this._html5FileName;
+        this._onLogFn?.([{ type: "normal", line }]);
+      }
+    });
+  };
+
+  private processLineCodes(line: string): boolean {
+    if (line.endsWith(LineCodes.RunEndCode)) {
+      // run completed
+      this.fetchResults();
+      return true;
+    }
+
+    if (line.includes(LineCodes.SessionCreatedCode)) {
+      this.storePassword();
+      return true;
+    }
+
+    if (line.includes(LineCodes.ResultsFetchedCode)) {
+      this.displayResults();
+      return true;
+    }
+
+    if (line.includes(LineCodes.RunCancelledCode)) {
+      this._runResolve({});
+      return true;
+    }
+
+    return false;
+  }
+
+  /**
+   * Generic call for use on stdin write completion.
+   * @param err The error encountered on the write attempt. Undefined if no error occurred.
+   */
+  private onWriteComplete = (err: Error): void => {
+    if (err) {
+      this._runReject?.(err);
+    }
+  };
+
+  /**
+   * Not implemented.
+   */
+  public sessionId = (): string => {
+    throw new Error("Not Implemented");
+  };
+
+  /**
+   * Fetches the ODS output results for the latest html results file.
+   */
+  private fetchResults = async () => {
+    const globalStorageUri = getGlobalStorageUri();
+    try {
+      await workspace.fs.readDirectory(globalStorageUri);
+    } catch (e) {
+      await workspace.fs.createDirectory(globalStorageUri);
+    }
+
+    this._pollingForLogResults = false;
+    const outputFileUri = Uri.joinPath(
+      globalStorageUri,
+      `${this._html5FileName}.htm`,
+    );
+    const directorySeparator =
+      this._workDirectory.lastIndexOf("/") !== -1 ? "/" : "\\";
+    const filePath =
+      this._config.protocol === ITCProtocol.COM
+        ? resolve(this._workDirectory, this._html5FileName + ".htm")
+        : `${this._workDirectory}${directorySeparator}${this._html5FileName}.htm`;
+    this._shellProcess.stdin.write(
+      `$filePath = "${filePath}"
+$outputFile = "${outputFileUri.fsPath}"
+$runner.FetchResultsFile($filePath, $outputFile)\n`,
+      this.onWriteComplete,
+    );
+  };
+
+  private displayResults = async () => {
+    const globalStorageUri = getGlobalStorageUri();
+    const outputFileUri = Uri.joinPath(
+      globalStorageUri,
+      `${this._html5FileName}.htm`,
+    );
+    const file = await workspace.fs.readFile(outputFileUri);
+
+    const htmlResults = (file || "").toString();
+    if (file) {
+      workspace.fs.delete(outputFileUri);
+    }
+
+    const runResult: RunResult = {};
+    if (htmlResults.search('<*id="IDX*.+">') !== -1) {
+      runResult.html5 = htmlResults;
+      runResult.title = "Result";
+    }
+    this._runResolve(runResult);
```

Comment:
Currently, if I uncheck `SAS.results.html.enabled` settings, simply run `%put foo;`. The running will never complete.
We need to also resolve the promise if no `this._html5FileName` captured, as ssh does
https://github.com/sassoftware/vscode-sas-extension/blob/main/client/src/connection/ssh/index.ts#L134

Reply:
Resolved in https://github.com/sassoftware/vscode-sas-extension/pull/592/commits/3b80433ddddc02f46f6b00c2e633547af8fdf477